packages/gestalt/src/Link.js

```diff
@@ -18,6 +18,59 @@ import getRoundingClassName from './getRoundingClassName.js';
 import focusStyles from './Focus.css';
 import useFocusVisible from './useFocusVisible.js';
 import layoutStyles from './Layout.css';
+import Icon from './Icon.js';
+import Box from './Box.js';
+import Text from './Text.js';
+
+type ExternalLinkIcon =
+  | 'none'
+  | 'default'
+  | {|
+      size: $ElementType<React$ElementConfig<typeof Text>, 'size'>,
+      color: $ElementType<React$ElementConfig<typeof Icon>, 'color'>,
+    |};
+
+function ExternalIcon({
+  externalLinkIcon,
+  position,
+}: {|
+  externalLinkIcon: ExternalLinkIcon,
+  position: 'rtl' | 'ltr',
+|}): Node {
+  const externalLinkIconMap = {
+    '100': 12,
+    '200': 14,
+    '300': 16,
+    '400': 20,
+    '500': 28,
+    '600': 36,
+    'sm': 12,
+    'md': 14,
+    'lg': 16,
+  };
+
+  return externalLinkIcon === 'none' ? null : (
```

Comment:
this check isn't necessary, you're already checking at the callsite (or keep this one and remove those)

Reply:
I know you were gonna point at this ones. 

I need them within ExternalIcon for Flow issues. I also put them outside cos I though it'd be more explicit for the parent to control the render or not, but needed the conditional inside as well for Flow.

I'll keep it inside as it's needed.

Reply:
DONE

Reply:
Makes sense, Flow needs some convincing sometimes. But consider leaving a comment (in the code) in those situations, so Future Us doesn't waste time trying to refactor it once we've forgotten the context